src/main/java/app/oengus/entity/model/User.java

```diff
@@ -136,7 +159,7 @@ public boolean isAccountNonExpired() {
     @Override
     @JsonIgnore
     public boolean isAccountNonLocked() {
-        return false;
+        return !roles.contains(Role.ROLE_BANNED);
     }
```

Comment:
Double negation, following would reduce cognitive complexity:
```
public boolean isAccountLocked() {
    return roles.contains(Role.ROLE_BANNED);
}
```

Reply:
Unfortunately, this method comes from the spring framework. I can add that second method but this override is required